WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -757,12 +757,17 @@ - (void)showPostStatusSelector
         [titles addObject:[BasePost titleForStatus:status]];
     }
 
+    NSString *postStatusTitle = @"";
+    if(self.apost.status.length) {
+        postStatusTitle = self.apost.statusTitle;
+    }
+       
     NSDictionary *statusDict = @{
                                  @"DefaultValue": NSLocalizedString(@"Published", @""),
                                  @"Title" : NSLocalizedString(@"Status", nil),
                                  @"Titles" : titles,
-                                 @"Values" : statuses,
-                                 @"CurrentValue" : self.apost.status
+                                 @"Values" : titles,
+                                 @"CurrentValue" : postStatusTitle
```

Comment:
This change will result in the title of a status (e.g. Published, Pending Review, Draft) being assigned where a status const value (e.g. publish, pending, draft) is expected.